pkg/api/http_server.go

```diff
@@ -74,12 +74,17 @@ func (hs *HttpServer) Start(ctx context.Context) error {
 			return nil
 		}
 	case setting.SOCKET:
-		ln, err := net.Listen("unix", setting.SocketPath)
+		ln, err := net.ListenUnix("unix", &net.UnixAddr{Name: setting.SocketPath, Net: "unix"})
 		if err != nil {
 			hs.log.Debug("server was shutdown gracefully")
 			return nil
 		}
 
+		ln.SetUnlinkOnClose(true)
```

Comment:
Could you remove this line? true is the default for unlinkOnClose when the socket is created via net.ListenUnix.